### PR TITLE
[ #53 , JSON output option ]

### PR DESCRIPTION
**Pull request by mohamedsalem401**
Write output `fileObjects` to .markdowndb/files.json  # array of file objects

### CODE REVIEW

File: src/lib/markdowndb.ts [hunk 1]
```diff
@@ -191,3 +193,17 @@ export class MarkdownDB {
     this.db.destroy();
   }
 }
+
```
Comment (rufuspollock) [inline src/lib/markdowndb.ts:196]: Can we get function documentation for important public functions like that ... (jsdoc format would be nice).

**Review by rufuspollock — CHANGES_REQUESTED** [review]
I'd think we'd want a bit of docs here - both in README.md plus probably on the command line to explain that this happens.

I'd have also kept the code path for generating json separate from sqlite both in code and in command line (i probably don't want both).

File: src/lib/markdowndb.ts (hunk 1 above)
Comment (mohamedsalem401) [reply src/lib/markdowndb.ts:196]: Okay, but I would like to create a new issue/pull request for that? I'd like to incorporate a few changes:
1. Include `strict: true`, as it is considered a best practice. This modification will necessitate some updates in the code but will substantially reduce the occurrence of bugs.
2. Initiate a discussion on how the library should behave on Windows versus macOS concerning `file_path`. Additionally, provide a fix for unit tests (as they currently fail on Windows due to path library differences)."
image

**Comment by rufuspollock** [comment]
> > I'd have also kept the code path for generating json separate from sqlite both in code and in command line (I probably don't want both).
> 
> You mean to add an option to generate either SQL or JSON but not both simultaneously? Right?

OK. I'm going to apply YAGNI and for now let's just generate both as you are doing. let's not bother with both paths.

File: src/lib/markdowndb.ts (hunk 1 above)
Comment (rufuspollock) [reply src/lib/markdowndb.ts:196]: OK.

**Comment by rufuspollock** [comment]
@mohamedsalem401 the one thing to do is probably just to add a changeset and then we can merge as is for now.

**Pull request merged**